deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+                    threshold = tree_partitioner.tree_.threshold[0]
+                    color_col = data[feature].ge(threshold)
+
+                    segment1_text, segment1_details = get_segment_details(model, scorer, test_dataset, data,
+                                                                          color_col)
+                    segment2_text, segment2_details = get_segment_details(model, scorer, test_dataset, data,
+                                                                          ~color_col)
+                    color_col = color_col.replace([True, False], [segment1_text, segment2_text])
+                    if segment1_details['score'] >= segment2_details['score']:
+                        color_map = {segment1_text: ok_color, segment2_text: weak_color}
+                    else:
+                        color_map = {segment1_text: weak_color, segment2_text: ok_color}
+                else:
+                    color_col = data[error_col_name]
+                    color_map = None
+                display.append(px.scatter(data, x=feature, y=error_col_name, color=color_col,
+                                          title=f'Segmentation of error by {feature}',
+                                          labels={error_col_name: 'model error'},
+                                          color_discrete_map=color_map))
+
+            value[feature] = {'segment1': segment1_details or None, 'segment2': segment2_details or None}
+
+            display[-1].update_layout(width=1200, height=400)
+
+        headnote = """<span>
+            The following graphs show the distribution of error for top features that are most useful for distinguishing
+            high error samples from low error samples.
+        </span>"""
+        display = [headnote] + display if display else None
+
+        return CheckResult(value, display=display)
+
+    def add_condition_segments_ratio_performance_change_not_greater_than(self, max_ratio_change: float = 0.05):
+        """Add condition - require that the difference of performance between the segments does not exceed a ratio.
+
+        Args:
+            max_ratio_change (float): maximal ratio of change between the two segments' performance.
+        """
+
+        def condition(result: Dict) -> ConditionResult:
+            fails = []
+            for feature in result.keys():
+                # If only one segment identified
+                if len(result[feature]) < 2:
+                    continue
+                performance_diff = \
+                    abs(result[feature]['segment1']['score'] - result[feature]['segment2']['score']) / \
+                    abs(max(result[feature]['segment1']['score'], result[feature]['segment2']['score']))
+                if performance_diff > max_ratio_change:
+                    fails.append(feature)
+
+            if fails:
+                msg = f'Segmentation of error by the features: {", ".join(sorted(fails))} resulted in percent change' \
```

Comment:
percent change is a weird phrase IMO. "change percentage"?
Would change the phrasing here to maybe "Segmentation of error by the features [] resulted in change in {} larger than {}%". Although that's also not that great. Maybe we can try f2f if you agree with me